src/hooks/__internal__/useFocusPortalContent/useFocusPortalContent.tsx

```diff
@@ -0,0 +1,82 @@
+import { RefObject, useEffect } from "react";
+import Events from "../../../__internal__/utils/helpers/events";
+import { defaultFocusableSelectors } from "../../../__internal__/focus-trap/focus-trap-utils";
+
+export default (
+  container: RefObject<HTMLElement>,
+  target: RefObject<HTMLElement>,
+  focusCallback?: (ev: React.KeyboardEvent<HTMLElement>) => void
+) => {
+  useEffect(() => {
+    const handleFocusPortalContent = (ev: KeyboardEvent) => {
+      const focusableElementsInContainer = Array.from(
+        container.current?.querySelectorAll(defaultFocusableSelectors) ||
+          /* istanbul ignore next */ []
+      );
+
+      if (target?.current === document.activeElement) {
+        if (
+          Events.isTabKey(ev) &&
+          !Events.isShiftKey(ev) &&
+          focusableElementsInContainer[0]
+        ) {
+          ev.preventDefault();
+          (focusableElementsInContainer[0] as HTMLElement)?.focus();
+        }
+
+        return;
+      }
+
+      /* istanbul ignore if */
+      if (!focusableElementsInContainer.length) {
+        return;
+      }
+
+      const lastElementFocused =
+        focusableElementsInContainer[
+          focusableElementsInContainer.length - 1
+        ] === document.activeElement;
+      const firstElementFocused =
+        focusableElementsInContainer[0] === document.activeElement;
+
+      if (Events.isTabKey(ev)) {
+        // last element focused inside portal navigate to next element in DOM aftet the target/ trigger element
+        if (lastElementFocused && !Events.isShiftKey(ev)) {
+          ev.preventDefault();
+
+          const allFocusableElements = Array.from(
+            document.querySelectorAll(defaultFocusableSelectors) ||
+              /* istanbul ignore next */ []
+          ) as HTMLElement[];
+          const filteredElements = allFocusableElements.filter(
+            (el) => el === target.current || Number(el.tabIndex) !== -1
+          );
+          const nextIndex =
+            filteredElements.indexOf(target.current as HTMLElement) + 1;
+
+          focusCallback?.((ev as unknown) as React.KeyboardEvent<HTMLElement>);
```

Comment:
Not a huge fan of this but I couldn't think of a better way to coerce/ get the types to line up here

Reply:
I actually realised that the `onClose` callback accepts `Event` as a type for the argument so I can just go with 

```
focusCallback: (ev: KeyboardEvent) => void;